### PR TITLE
test(node): improve unit test coverage for node module

### PR DESCRIPTION
**Pull request by sanity**
## Problem

The `crates/core/src/node` module had low unit test coverage, particularly `mod.rs` at 3.34% and `network_bridge.rs` at 80.37%. This makes it harder to catch regressions and understand expected behavior of core node components.

## This Solution

Add comprehensive unit tests focusing on testable pure logic without requiring complex network mocking:

### Coverage Improvements

| File | Before | After | Improvement |
|------|--------|-------|-------------|
| `node/mod.rs` | 3.34% | 23.19% | **+19.85%** |
| `node/network_bridge.rs` | 80.37% | 89.05% | **+8.68%** |
| `node/network_bridge/priority_select.rs` | 60.45% | 76.87% | **+16.42%** |

### Tests Added

**`node/mod.rs`**:
- `PeerId` tests (equality based on address, ordering, hashing, serialization, display)
- `InitPeerNode` construction tests  
- `is_operation_completed` function with various error conditions
- Hostname resolution tests (including edge cases like trailing dots, invalid ports)

**`node/network_bridge.rs`**:
- `ConnectionError` variant tests (clone, display, debug, from conversions)
- Channel notification flow tests
- Channel capacity under load tests
- Sender cloning behavior tests

### Files Still Needing Coverage

The following files have 0% coverage and would benefit from future test work:
- `p2p_impl.rs` - Requires network mocking
- `p2p_protoc.rs` - Requires libp2p protocol mocking  
- `in_memory.rs` - Test infrastructure code

These are harder to test in isolation because they involve real network I/O and complex async coordination. The existing integration tests exercise much of this code.

## Testing

All 78 node module tests pass:
```
cargo test --lib 'node::' -- --test-threads=4
test result: ok. 78 passed; 0 failed; 0 ignored
```

[AI-assisted - Claude]

### CODE REVIEW

File: crates/core/src/node/network_bridge.rs [hunk 1]
```diff
@@ -283,4 +283,156 @@ mod tests {
...
+    }
+}
+
+// ConnectionError tests
+#[cfg(test)]
+mod connection_error_tests {
+    use super::*;
+
+    #[test]
+    fn test_connection_error_clone() {
```
Comment (iduartgomez) [inline crates/core/src/node/network_bridge.rs:364]: idk if I fiund much utility in this kind of test tbh

Comment (sanity) [reply crates/core/src/node/network_bridge.rs:364]: Any of them or just this one?
